src/database/node.rb

```diff
@@ -41,11 +41,12 @@ def move_to(position)
 
   # Moves all connected components
   # this is very slow. Only do this if necessary (i.e. not in simulation)
+  # It is important to first update the hubs, as some edges will use the hub
   def update_sketchup_object
-    @incidents.each(&:update_sketchup_object)
-    @adjacent_triangles.each(&:update_sketchup_object)
     pods.each { |pod| pod.update_position(@position) }
     hub.entity.move!(Geom::Transformation.new(@position))
+    @incidents.each(&:update_sketchup_object)
+    @adjacent_triangles.each(&:update_sketchup_object)
```

Comment:
Did you check that this doesn't break the conventional physics simulation?

Reply:
It does not seems so, although im not sure if it works in all potential cases